src/ClusterManager/job_manager.py

```diff
@@ -591,19 +595,39 @@ def mark_schedulable_non_preemptable_jobs(jobs_info, cluster_schedulable,
         if preemption_allowed:
             continue # schedule non preemptable first
 
-        if cluster_schedulable >= job_resource and \
+        if job_info["status"] in ["scheduling", "running"]:
+            job_info["allowed"] = True # do not preempt non preemptable jobs
+            vc_schedulable -= job_resource
+            cluster_schedulable -= job_resource
+            continue
+
+        if stop_scheduling is None and \
+                cluster_schedulable >= job_resource and \
                 vc_schedulable >= job_resource:
             vc_schedulable -= job_resource
             cluster_schedulable -= job_resource
             job_info["allowed"] = True
-            logger.info("Allow non-preemptable job %s to run, job resource %s",
-                        job_id, job_resource)
-        else:
+            logger.info("Allow non-p job %s to run, job resource %s", job_id,
```

Comment:
`"Allow non-preemptable job"`

Reply:
Maybe add vc info in the logging as well?

Reply:
I wanted to make log shorter.